vllm/core/scheduler.py

```diff
@@ -83,17 +84,25 @@ def __init__(
         # LoRAs. This should be improved in the future.
         self.lora_config = lora_config
 
-        self.prompt_limit = min(self.scheduler_config.max_model_len,
-                                self.scheduler_config.max_num_batched_tokens)
+        if self.scheduler_config.embedding_mode:
+            # Profile this value if necessary
+            self.prompt_limit = self.scheduler_config.max_num_batched_tokens
+        else:
+            self.prompt_limit = min(
+                self.scheduler_config.max_model_len,
+                self.scheduler_config.max_num_batched_tokens)
 
         # Instantiate the scheduling policy.
         self.policy = PolicyFactory.get_policy(policy_name="fcfs")
-        # Create the block space manager.
-        self.block_manager = BlockSpaceManager(
-            block_size=self.cache_config.block_size,
-            num_gpu_blocks=self.cache_config.num_gpu_blocks,
-            num_cpu_blocks=self.cache_config.num_cpu_blocks,
-            sliding_window=self.cache_config.sliding_window,
+        # Decide on the block manager class based on embedding mode
+        block_manager_class = (DummyBlockSpaceManager
+                               if self.scheduler_config.embedding_mode else
+                               BlockSpaceManager)
+        self.block_manager = block_manager_class(
```

Comment:
This thing s so deeply coupled with the scheduler. Can't think of a easy way to do if-then

Reply:
See comment above. Changed it to ProxyPattern. This if-else is removed.